org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/SingletonClassAnnotation.java

```diff
@@ -16,35 +16,34 @@
 /**
  * A class annotation configuration for the <code>@Singleton</code> annotation.
  */
-@SuppressWarnings("all")
 public class SingletonClassAnnotation implements IClassAnnotation {
-  @Override
-  public CharSequence generate() {
-    return "@Singleton";
-  }
-  
-  @Override
-  public boolean appliesTo(final JavaFileAccess javaFile) {
-    return true;
-  }
-  
-  @Override
-  public TypeReference getAnnotationImport() {
-    return new TypeReference(Singleton.class);
-  }
-  
-  @Override
-  public String toString() {
-    return this.generate().toString();
-  }
-  
-  @Override
-  public boolean equals(final Object obj) {
-    return (obj instanceof SingletonClassAnnotation);
-  }
-  
-  @Override
-  public int hashCode() {
-    return this.getClass().getName().hashCode();
-  }
+	@Override
+	public CharSequence generate() {
+		return "@Singleton";
```

Comment:
This will break when there is a simple name clash.

Reply:
yes but it always did